lib/src/scanner.dart

```diff
@@ -160,18 +171,28 @@ class NgScanner {
       _state = NgScannerState.scanStart;
       return new NgToken.commentEnd(_current.offset);
     }
-    return handleError();
+    // Only triggered by EOF.
+    return handleError(
+      NgParserWarningCode.AFTER_COMMENT,
+      _lastCommentStartOffset,
+      _current.offset - _lastCommentStartOffset,
+    );
   }
 
+  // TODO: Max: Better handle cases like 'prop]'. Instead of
+  // TODO: 'prop []' resolve, resolve as '[prop].
```

Comment:
Sure, we can move this to the parser. The only concern I have with this is that it adds a lot of extra overhead into the primary converted NodeInfos we use in angular_analyzer_plugin (current implementation takes the overall AST and uses it at face value; doesn't dissect the individual components for synthetic parts).